tests/test_salesforce_lookback_window.py

```diff
@@ -0,0 +1,108 @@
+from base import SalesforceBaseTest
+
+class SalesforceLookbackWindow(SalesforceBaseTest):
+
+    # subtract the desired amount of seconds form the date and return
+    def timedelta_formatted(self, dtime, seconds=0):
+        try:
+            date_stripped = datetime.strptime(dtime, self.START_DATE_FORMAT)
+            return_date = date_stripped + timedelta(seconds=seconds)
+
+            return datetime.strftime(return_date, self.START_DATE_FORMAT)
+
+        except ValueError:
+            try:
+                date_stripped = datetime.strptime(dtime, self.BOOKMARK_COMPARISON_FORMAT)
+                return_date = date_stripped + timedelta(seconds=seconds)
+
+                return datetime.strftime(return_date, self.BOOKMARK_COMPARISON_FORMAT)
+
+            except ValueError:
+                return Exception("Datetime object is not of the format: {}".format(self.START_DATE_FORMAT))
+
+    @staticmethod
+    def name():
+        return 'tap_tester_salesforce_lookback_window'
+
+    @staticmethod
+    def get_properties():  # pylint: disable=arguments-differ
+        return {
+            'start_date' : '2021-11-10T00:00:00Z',
+            'instance_url': 'https://singer2-dev-ed.my.salesforce.com',
+            'select_fields_by_default': 'true',
+            'api_type': 'REST',
+            'is_sandbox': 'false',
+            'lookback_window': 86400
+        }
+
+    def expected_sync_streams(self):
+        return {
+            'Account'
+        }
+
+    def test_run(self):
+        # create connection
+        conn_id = connections.ensure_connection(self)
+        # create state file
+        state = {
+            'bookmarks':{
+                'Account': {
+                    'SystemModstamp': '2021-11-10T00:00:00.000000Z'
+                }
+            }
+        }
+        # set state file to run in sync mode
+        menagerie.set_state(conn_id, state)
+
+        # run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # select certain catalogs
+        expected_streams = self.expected_sync_streams()
+        catalog_entries = [catalog for catalog in found_catalogs
+                            if catalog.get('tap_stream_id') in expected_streams]
+
+        # stream and field selection
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+
+        # run sync
+        self.run_and_verify_sync(conn_id)
+
+        # get synced records
+        sync_records = runner.get_records_from_target_output()
+
+        # get replication keys
+        expected_replication_keys = self.expected_replication_keys()
+
+        # get start date ie. date from which the sync started
+        start_date = state.get('bookmarks').get('Account').get('SystemModstamp')
+        # calculate the simulated start date by subtracting lookback window seconds
+        start_date_with_lookback_window = self.timedelta_formatted(start_date, seconds=-self.get_properties()['lookback_window'])
```

Comment:
Can you change the wording so that we do not mention start date? That term should be reserved for historical syncs. The point at which the sync starts on an incremental sync is based on the state you have set above, not the start date. Maybe I'm just being picky but I think it could be confusing to a tester in the future.

Reply:
Updated the variable name.